spec/models/api/v1/query_param_spec.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe Api::V1::QueryParam do
+  let(:dummy_controller) { Api::V1::BaseController.new.extend Api::V1::QueryParam }
+
+  describe "#invalid_query_param" do
+    it "renders error" do
+      allow(dummy_controller).to receive(:render) {}
+      dummy_controller.invalid_query_param("param", :unprocessable_entity, "error message")
+      expect(dummy_controller).to have_received(:render).with(
+        json:
+          {
+            errors:
+            [{
+              code: 422,
+              detail: "error message",
+              source: { parameter: "param" },
+              status: :unprocessable_entity,
+              title: I18n.t("api.query_param.error.title")
```

Comment:
Better to spell it out in plain text. https://github.com/openfoodfoundation/openfoodnetwork/wiki/Code-Conventions#prefer-plain-text-over-method-calls-in-expected-values

Reply:
Fixed